src/Serialization/main.jl

```diff
@@ -123,6 +139,14 @@ function save_type_dispatch(s::SerializerState, obj::T) where T
     return result
 end
 
+function load_type_dispatch(s::DeserializerState,
+                            ::Type{T}, str::String; parent=nothing) where T <: OscarBasicType
```

Comment:
Here we can just do this to eliminate the need for `OscarBasicType, as the signature already is unique due to `str` instead of `dict` being passed
```suggestion
                            ::Type{T}, str::String; parent=nothing) where T
    @assert is_basic_serialization_type(T)
```
The assertion is of course just for paranoia